great_expectations/datasource/data_connector/data_connector.py

```diff
@@ -316,30 +318,37 @@ def _self_check_fetch_batch(
         if pretty_print:
             print(f"\n\t\tFetching batch data...")
 
-        batch_definition_list = self._map_data_reference_to_batch_definition_list(
+        batch_definition_list: List[
+            BatchDefinition
+        ] = self._map_data_reference_to_batch_definition_list(
             data_reference=example_data_reference,
             data_asset_name=data_asset_name,
         )
         assert len(batch_definition_list) == 1
-        batch_definition = batch_definition_list[0]
+        batch_definition: BatchDefinition = batch_definition_list[0]
 
         # _execution_engine might be None for some tests
         if batch_definition is None or self._execution_engine is None:
             return {}
+
+        batch_data: Any
+        batch_spec: BatchSpec
```

Comment:
Yep! It's pretty new to me too. Been learning a lot of good new habits from @alexsherstinsky (who's great at typehinting). In this case, it's useful because those variables get assigned values below using multiple assignment/unpacking.